File: VB-Examples/24_Workbook/OpenFiles.vb
Imports System.Collections
Imports System.ComponentModel

Imports Spire.Xls
Imports System.IO
Imports System.Text

Namespace OpenFiles
	Partial Public Class Form1
		Inherits Form
		Public Sub New()
			InitializeComponent()
		End Sub
		Private Sub btnRun_Click(ByVal sender As Object, ByVal e As EventArgs) Handles btnRun.Click
            ' Declare file paths for different types of files
            Dim filepath As String = "..\..\..\..\..\..\Data\ExcelSample_N1.xlsx" ' Path to Excel file
            Dim filepath97 As String = "..\..\..\..\..\..\Data\ExcelSample97_N.xls" ' Path to Excel 97-2003 file
            Dim filepathXml As String = "..\..\..\..\..\..\Data\OfficeOpenXML_N.xml" ' Path to XML file
            Dim filepathCsv As String = "..\..\..\..\..\..\Data\CSVSample_N.csv" ' Path to CSV file

            ' Create a StringBuilder object to store messages
            Dim builder As New StringBuilder()

            ' Open and load workbook from the specified Excel file path
            Dim workbook1 As New Workbook()
            workbook1.LoadFromFile(filepath)
            builder.AppendLine("Workbook opened using file path successfully!")

            ' Open and load workbook from the specified file stream
            Dim stream As New FileStream(filepath, FileMode.Open)
            Dim workbook2 As New Workbook()
            workbook2.LoadFromStream(stream)
            builder.AppendLine("Workbook opened using file stream successfully!")
            stream.Dispose()

            ' Open and load an Excel 97-2003 workbook
            Dim wbExcel97 As New Workbook()
            wbExcel97.LoadFromFile(filepath97, ExcelVersion.Version97to2003)
            builder.AppendLine("Microsoft Excel 97 - 2003 workbook opened successfully!")

            ' Open and load an XML file as a workbook
            Dim wbXML As New Workbook()
            wbXML.LoadFromXml(filepathXml)
            builder.AppendLine("XML file opened successfully!")

            ' Open and load a CSV file as a workbook, specifying delimiter and start row/column
            Dim wbCSV As New Workbook()
            wbCSV.LoadFromFile(filepathCsv, ",", 1, 1)
            builder.AppendLine("CSV file opened successfully!")

            ' Write the contents of the StringBuilder to a text file
            Dim output As String = "OpenFiles_out.txt"
            File.WriteAllText(output, builder.ToString())

            ' Release the resources used by the workbook1
            workbook1.Dispose()

            ' Release the resources used by the workbook2
            workbook2.Dispose()

            ' Release the resources used by the wbExcel97
            wbExcel97.Dispose()

            ' Release the resources used by the wbXML
            wbXML.Dispose()

            ' Release the resources used by the wbCSV
            wbCSV.Dispose()

            'Launch the file
            ExcelDocViewer(output)
		End Sub
		Private Sub ExcelDocViewer(ByVal fileName As String)
			Try
				Process.Start(fileName)
			Catch
			End Try
		End Sub
		Private Sub btnClose_Click(ByVal sender As Object, ByVal e As EventArgs) Handles btnClose.Click
			Close()
		End Sub

	End Class
End Namespace

File: CS-Examples/24_Workbook/OpenFiles.cs
using System;
using System.Drawing;
using System.Collections;
using System.ComponentModel;
using System.Windows.Forms;
using System.Data;

using Spire.Xls;
using System.IO;
using System.Text;

namespace OpenFiles
{
    public partial class Form1 : Form
    {
        public Form1()
        {
            InitializeComponent();
        }
        private void btnRun_Click(object sender, System.EventArgs e)
        {
             // Pathes of files
            string filepath = @"..\..\..\..\..\..\Data\ExcelSample_N1.xlsx";
            string filepath97 = @"..\..\..\..\..\..\Data\ExcelSample97_N.xls";
            string filepathXml = @"..\..\..\..\..\..\Data\OfficeOpenXML_N.xml";
            string filepathCsv = @"..\..\..\..\..\..\Data\CSVSample_N.csv";

            // Create string builder
            StringBuilder builder = new StringBuilder();

            // 1. Load file by file path
            // Create a workbook
            Workbook workbook1 = new Workbook();
            // Load the document from disk
            workbook1.LoadFromFile(filepath);
            builder.AppendLine("Workbook opened using file path successfully!");

            // Dispose of the workbook object to release resources 
            workbook1.Dispose();

            // 2. Load file by file stream
            FileStream stream = new FileStream(filepath, FileMode.Open);

            // Create a workbook
            Workbook workbook2 = new Workbook();

            // Load the document from disk
            workbook2.LoadFromStream(stream);
            builder.AppendLine("Workbook opened using file stream successfully!");
            stream.Dispose();

            // Dispose of the workbook object to release resources 
            workbook2.Dispose();

            // 3. Open Microsoft Excel 97 - 2003 file
            Workbook wbExcel97 = new Workbook();
            wbExcel97.LoadFromFile(filepath97, ExcelVersion.Version97to2003);
            builder.AppendLine("Microsoft Excel 97 - 2003 workbook opened successfully!");

            // 4. Open xml file
            Workbook wbXML = new Workbook();
            wbXML.LoadFromXml(filepathXml);
            builder.AppendLine("XML file opened successfully!");

            // Dispose of the workbook object to release resources 
            wbExcel97.Dispose();

            //5. Open csv file
            Workbook wbCSV = new Workbook();
            wbCSV.LoadFromFile(filepathCsv, ",", 1, 1);
            builder.AppendLine("CSV file opened successfully!");

            //Save to txt file
            string output = "OpenFiles_out.txt";
            File.WriteAllText(output, builder.ToString());

            // Dispose of the workbook object to release resources 
            wbCSV.Dispose();

            // Launch the file
            ExcelDocViewer(output);
        }
        private void ExcelDocViewer(string fileName)
        {
            try
            {
                System.Diagnostics.Process.Start(fileName);
            }
            catch { }
        }
        private void btnClose_Click(object sender, EventArgs e)
        {
            Close();
        }

    }
}
